### DISCUSSION TITLE
Running Laragon at system startup, not user login

### DISCUSSION THREAD

**Discussion by dleigh** [leokhoa/laragon#847] [Q&A]
Hi, 
In the Laragon preferences, I have checked:

- run Laragon when Windows starts
- Run minimized
- Start All automatically

For me, I would have assumed that this meant that a fully booted system with no one logged in would still serve the site that I have running on Laragon (presumably with some "system" credentials or credentials I set up beforehand).  Unfortunately, my experience is that the user has to be logged in for Laragon to run and serve the site. 

Is there a way to do this?

Any help is appreciated.
Thanks!

**Comment by ppfoong** [comment]
You can try out the following solution:

https://www.iceflatline.com/2015/12/run-a-windows-application-as-a-service-with-srvany/

**Comment by dleigh** [comment]
Thank you.  I'll try this out.

**Comment by dleigh** [comment]
Ok, srvany is part of the 2003 Resource Kit which can no longer be found at Microsoft.  I installed this: [http://nssm.cc/](http://nssm.cc/) which says that it's a drop-in replacement for srvany.  I set up a new user (administrative level) to run Laragon at system startup (I tried also with simple system-level access).  Neither worked.

This is what is logged: 

```
Faulting application name: laragon.exe, version: 6.0.0.2, time stamp: 0x63249b9e
Faulting module name: KERNELBASE.dll, version: 10.0.22621.3958, time stamp: 0xdac14c46
Exception code: 0x0eedfade
Fault offset: 0x0014a942
Faulting process id: 0x0x1CE4
Faulting application start time: 0x0x1DAE4299409AC77
Faulting application path: C:\laragon\laragon.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 650795f0-a549-4358-a24e-6004d19435b9
Faulting package full name: 
Faulting package-relative application ID: 
```
So I'm stuck still.  Has anyone successfully been able to host a site using Laragon that only required that the system be booted - NOT that any user be logged in?

Perhaps I'm going about this incorrectly.  Using the binaries/config files provided by the Laragon installation, should I just be running Apache and MySQL as services instead of trying to start up Laragon?  Personally, I'd prefer to use Laragon, but if that's doomed to failure I suppose another approach should be used?  I suppose also that what I'm asking to do, but not phrasing it well, is to run Laragon "headless".

Thanks!

**Comment by ppfoong** [comment]
Perhaps you can directly run httpd and mysqld inside the laragon\bin folder.

**Comment by dleigh** [comment]
Ok, this ended up being easier than I thought.  Hopefully this will help others. Here are my high-level steps for developing locally with Laragon and then having what I developed run headless on a normal Windows (11) machine:

1. Did all my normal Laragon setup and config and got my system running with all the right versions of MySQL and PHP, etc.  BUT I still had to be logged in to make it work.
2. I did **NOT** make Laragon run as a service, but took @ppfoong 's advice to **make Apache and MySQL run as services at system startup**.  I **unchecked** the boxes in the Laragon preferences about automatically starting those services.  Effectively, while still using the Laragon installation files and the web site files, _I'm not using Laragon anymore for serving the site_.  I suppose, as my site evolves, and I want to evolve the versions of PHP and MySQL, I could return to Laragon and use it for those purposes of changing versions, etc. (for which it's quite handy). 
3. To start Apache as a service (httpd daemon) I followed this video: [https://youtu.be/yZsvSgXuPiE?si=fsAc_0qj7loYddiy](https://youtu.be/yZsvSgXuPiE?si=fsAc_0qj7loYddiy)
4. To start MySQL as a service (mysqld daemon), I followed this web page: [https://dev.mysql.com/doc/refman/8.4/en/windows-start-service.html](https://dev.mysql.com/doc/refman/8.4/en/windows-start-service.html)

Now when I start the machine, Apache and MySQL start up automatically and no one has to be logged in, and I can access the site that's being served remotely. 

Thanks for the help @ppfoong !
